src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -127,27 +127,38 @@ def __init__(
                 num_heads=[4, 8, 16, 32],
                 window_size=12,
                 drop_path_rate=0.3,
+                out_features=["stage1", "stage2", "stage3", "stage4"],
             )
         else:
-            backbone_model_type = backbone_config.pop("model_type")
+            # verify that the backbone is supported
+            backbone_model_type = (
+                backbone_config.pop("model_type") if isinstance(backbone_config, dict) else backbone_config.model_type
+            )
             if backbone_model_type not in self.backbones_supported:
                 raise ValueError(
                     f"Backbone {backbone_model_type} not supported, please use one of"
                     f" {','.join(self.backbones_supported)}"
                 )
-            backbone_config = AutoConfig.for_model(backbone_model_type, **backbone_config)
+            if isinstance(backbone_config, dict):
+                config_class = CONFIG_MAPPING[backbone_model_type]()
```

Comment:
```suggestion
                config_class = CONFIG_MAPPING[backbone_model_type]
```
You shouldn't instantiate the class here.